### PR TITLE
MQTT File Download Agent

### PR DESCRIPTION
**Pull request by awsblake**
*Issue #, if available:*

*Description of changes:*
1. File download agent library
2. Download agent sample app
3. unit and integration tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws_iot_download_agent.h [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+ * AWS IoT services provided a feature for streaming contents of files
+ * over the MQTT protocol, in a block by block manner. The files are
+ * stored in the Simple Storage Service (S3) buckets of AWS. AWS provided
+ * APIs and Command Line Interface (CLI) for creating streams in the cloud
+ * and attaching files to the streams. See the README.md file under the
+ * samples/linux/download_agent_sample directory for instructions.
+ *
+ * This library provides a simple API that can be integrated on an IoT
+ * device for interacting with AWS IoT services and downloading files.
+ * A device can integrate this libraryct as long as it has the capability
```
[the remark this answers is not in the document]
Comment (awsblake) [reply include/aws_iot_download_agent.h]: Fixed by 52cdc28

File: include/aws_iot_download_agent.h [hunk 2]
```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
```
[the remark this answers is not in the document]
Comment (awsblake) [reply include/aws_iot_download_agent.h]: Fixed by 52cdc28

File: include/aws_iot_download_cbor.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * http://www.FreeRTOS.org
+ */
+
+#ifndef __AWS_OTACBOR__H__
+#define __AWS_OTACBOR__H__
+
+/**
+ * @brief Decode a Get Stream response message from AWS IoT OTA.
+ */
+int OTA_CBOR_Decode_GetStreamResponseMessage( const uint8_t * pucMessageBuffer,
```
[the remark this answers is not in the document]
Comment (awsblake) [reply include/aws_iot_download_cbor.h]: This PR is not related to OTA, replace all the OTA wording in function, macros and comments by 52cdc28.

File: README.md [hunk 4]
```diff
@@ -34,6 +34,9 @@ The Device SDK implements the specific protocol for Thing Shadows to retrieve, u
 ### Jobs
 The Device SDK implements features to facilitate use of the AWS Jobs service. The Jobs service can be used for device management tasks such as updating program files, rotating device certificates, or running other maintenance tasks such are restoring device settings or restarting devices.
 
+### MQTT Download Agent
+The Device SDK implements a file download agent library that can be integrated any MQTT stack that’s compatible with AWS IoT. This file download agent interacts with the [Block Transfer (Streaming) Service of AWS IoT](https://docs.aws.amazon.com/iot/latest/apireference/API_CreateStream.html). It downloads files block by block over the MQTT protocol.
```
[the remark this answers is not in the document]
Comment (awsblake) [reply README.md]: Fixed by d751452

**Comment by alfred2g** [comment]
Closing this pull request as we are currently only accepting critical bug fixes in v3 SDK. Any new development should happen on v4 SDK.

**Pull request closed without merging**